### ISSUE TITLE
How to encrypt with a public key which are protected by a passphrase?

### ISSUE THREAD

**Issue by leejie8008** [singpolyma/openpgp-php#75]
I am trying to encrypt and decrypt with public and private key which are protected by a passphrase. The decrypt result is always showing `null`, and it might be because that the keys are protected by passphrase.

Below are my implementation for the encrypt and decryption. I get `Encryption Error` at line `$decrypted = $decryptor->decrypt($encrypted);` with this implementation

`$message = 'Test Text';`

`$pathToYourAsciiArmoredPublicKey = base_path('public.asc');`
`$pathToYourAsciiArmoredPrivateKey = base_path('private.asc');`

`$key=OpenPGP_Message::parse(OpenPGP::unarmor(file_get_contents($pathToYourAsciiArmoredPublicKey),'PGP PUBLIC KEY BLOCK'));`

`$data = new \OpenPGP_LiteralDataPacket($message);`

`$encrypted = \OpenPGP_Crypt_Symmetric::encrypt($key,new OpenPGP_Message(array($data)));`
`
$encryptedPrivateKey=OpenPGP_Message::parse(OpenPGP::unarmor(file_get_contents($pathToYourAsciiArmoredPrivateKey),'PGP PRIVATE KEY BLOCK'));`

`foreach ($encryptedPrivateKey as $package) {`
     `$key = OpenPGP_Crypt_Symmetric::decryptSecretKey('password', $package);`
     `$decryptor = new OpenPGP_Crypt_RSA($key);`
     `$decrypted = $decryptor->decrypt($encrypted);`
     `var_dump($decrypted);`
`}`

From what I found is, it already combine both together at `openpgp_crypt_symmetric.php`

` public static function encrypt($passphrases_and_keys, $message, $symmetric_algorithm=9) {`

Thanks!

**Comment by singpolyma** [comment]
Have you done `var_dump($key)` after the `decryptSecretKey` call to make sure it is decrypting correctly?

Also, is the key an RSA key?

**Comment by heavykevy** [comment]
@leejie8008,
I was able to use your code to decrypt my secret key and decrypt a packet with a password protected RSA key created by WinPGP.
One thing that I noticed that caused issues with your code was looping through all of the key packets.
Only the Private key packet should need to be decypted and used.
This is the code I used that worked for me:
$pkey = OpenPGP_Crypt_Symmetric::decryptSecretKey ( 'password', $encryptedPrivateKey->packets [0] );
var_dump ( $pkey );
$decryptor = new OpenPGP_Crypt_RSA ( $pkey );
$decrypted = $decryptor->decrypt ( $encrypted);

Best regards
